cclib/bridge/cclib2pyscf.py

```diff
@@ -112,4 +129,491 @@ def makepyscf_mos(ccdata, mol):
+
+    else:
+        attributes["metadata"]["unrestricted"] = False
+
+        nocc = [np.count_nonzero(scf.mo_occ != 0)]
+        attributes["moenergies"] = [scf.mo_energy]
+        attributes["homos"] = [nocc[0] - 1]
+        # Orbital coeffs.
+        attributes["mocoeffs"] = [scf.mo_coeff]
+
+    attributes["nmo"] = len(attributes["moenergies"][0])
+
+    # Orbital symmetries.
+    # if scf.mol.symmetry:
+    if hasattr(scf, "get_orbsym"):
+        # Symmetry labels are in scf.mol.irrep_name, symmetry ids are in scf.mol.irrep_id
+        symmetry_mapping = {
+            symm_id: symm_name for symm_id, symm_name in zip(scf.mol.irrep_id, scf.mol.irrep_name)
+        }
+
+        orbsyms = scf.get_orbsym() if scf.istype("UHF") else [scf.get_orbsym()]
+
+        attributes["mosyms"] = []
+        for alpha_beta in range(0, (len(attributes["moenergies"]))):
+            attributes["mosyms"].append(
+                [symmetry_mapping[symm_id] for symm_id in orbsyms[alpha_beta]]
+            )
+
+    # Dipole moments.
+    #
+    scf_density_matrix = scf.make_rdm1(scf.mo_coeff, scf.mo_occ)
+    # PySCF uses 0,0,0 as the origin, but this might not correspond to the centre of mass?
+    origin = (0, 0, 0)
+    attributes["moments"] = [
+        origin,
+        scf.dip_moment(scf.mol, scf_density_matrix, origin=origin, verbose=2),
+    ]
+
+    # Quadrupole moments are new (introduced ~ August 2024) and not yet widely available.
+    if hasattr(scf, "quad_moment"):
+        attributes["moments"].append(
+            scf.quad_moment(
+                scf.mol, scf_density_matrix, origin=origin, unit="DebyeAngstrom", verbose=2
+            )
+        )
+
+    # Mulliken.
+    mulliken_pop, mulliken_charges = scf.mulliken_pop(
+        scf.mol, scf_density_matrix, s=attributes["aooverlaps"], verbose=2
+    )
+    # PySCF describes this as 'Mulliken population analysis, based on meta-Lowdin AOs', is this what we want?
+    lowdin_pop, lowdin_charges = scf.mulliken_meta(
+        scf.mol, scf_density_matrix, s=attributes["aooverlaps"], verbose=2
+    )
+    attributes["atomcharges"] = {"mulliken": mulliken_charges, "lowdin": lowdin_charges}
+
+    # Excited states.
+    if len(et) > 0:
+        # PySCF tracks convergence for each state which is great.
+        attributes["metadata"]["success"] = all(
+            itertools.chain(*(etmethod.converged for etmethod in et))
+        )
+
+        # In cclib 1.x, 'energies' are actually expected to be cm-1. In 2.x, this will change to Hartree.
+        attributes["etenergies"] = list(itertools.chain(*(etmethod.e for etmethod in et)))
+        attributes["etoscs"] = list(
+            itertools.chain(*(etmethod.oscillator_strength(gauge="length") for etmethod in et))
+        )  # or do we want velocity?
+        # et.analyse() prints real symmetries, so they must be available somewhere...
+
+        attributes["etsyms"] = []
+        for etmethod in et:
+            if isinstance(etmethod, pyscf.tdscf.rks.TDA):
```

Comment:
I'd replace all this with a dict that maps `pyscf` class to the string, and on the dict use `.get(etmethod, type(etmethod).__name__)` to simultaneously handle the default case.